### ISSUE TITLE
"Go to definition" on external library code

### ISSUE THREAD

**Issue by ysangkok** [haskell/ghcide#745]
Apologies if the issue already exists, I am just submitting it from Reddit so that it doesn't get lost: [qualiaqq's post](https://www.reddit.com/r/haskell/comments/iipj3s/lets_brainstorm_what_plugins_would_you_like_to/g389dxd/)

**Comment by wz1000** [comment]
I have implemented this (along with references and other stuff) using https://github.com/wz1000/HieDb in this branch https://github.com/wz1000/ghcide/tree/hiedb-3 

See http://78.47.113.11/defs.gif

Instructions for use:

1. Generating `.hie` files for your dependencies by putting something like the following in your `cabal.project` file. (andequivalent method for stack should exist)

```
package *
  ghc-options: -fwrite-ide-info -hiedir /absolute/path/to/some/directory
```

2. Index the hie files by navigating to the root of your project directory (usually the directory with the `hie.yaml` file) and execute

```bash
$ ghcide hiedb index path/to/directory/with/hie/files
```

3. Enjoy go to definition for dependencies!

The branch is pretty stable at this point, but there still might be some rough spots. If you encounter any, please open an issue on my fork.

(This doesn't work for ghc boot libs since hie files for those a bit hard to come by unless you compile ghc yourself. I'm working to change this for ghc 9.0)

**Comment by jneira** [comment]
Closed as duplicate of #61 as it has more content (but i will link the awesome @wz1000 comment on how to use its incoming branch)
